From b3c2faf7ee519e4686248957ee079a2452741d61 Mon Sep 17 00:00:00 2001
From: Viktor Szakats <commit@vsz.me>
Date: Thu, 16 Jul 2026 13:05:51 +0200
Subject: [PATCH 2/2] build: constify more to fix gcc-16 / glibc 2.43 builds
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Upstream: https://github.com/curl/trurl/commit/b3c2faf7ee519e4686248957ee079a2452741d61
Applies unchanged to 0.16.1.

Fixing:
```
trurl.c:1869:12: error: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1869 |       sepw = strchr(w, ' ');
      |            ^
```
Ref: https://github.com/curl/trurl/actions/runs/29492873600/job/87602928996?pr=446#step:4:12

Reported-by: vpzomtrrfrt on github
Bug: https://github.com/curl/trurl/issues/430#issuecomment-4985712536
Follow-up to 6e1479cc3bdece8d9a7602e6f8f799305d5a5b7d #432

Closes #447
---
 trurl.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/trurl.c b/trurl.c
index f6e509f..9a5860e 100644
--- a/trurl.c
+++ b/trurl.c
@@ -325,7 +325,7 @@ struct iterinfo {
   CURLU *uh;
   const char *part;
   size_t plen;
-  char *ptr;
+  const char *ptr;
   unsigned int varmask; /* sets 1 << [component] */
 };
 
@@ -1811,7 +1811,7 @@ static void singleurl(struct option *o,
       const char *w;
       size_t wlen;
       const char *sep;
-      char *sepw;
+      const char *sepw;
       bool urlencode = true;
       const struct var *v;
 
